runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7550,7 +7546,18 @@ TR::CompilationInfoPerThreadBase::postCompilationTasks(J9VMThread * vmThread,
 #endif /* defined(JITSERVER_SUPPORT) */
 
    if (_compiler)
+      {
+      bool isToFreeKnownObjectTable = true;
+      // Compilation aborted early at the server
+      if (_compiler->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER && !metaData)
+         isToFreeKnownObjectTable = false;
+
+      // The KOT needs to survive at least until we're done committing virtual guards
+      if (_compiler->getKnownObjectTable() && isToFreeKnownObjectTable)
+         _compiler->freeKnownObjectTable();
+
```

Comment:
This change will be removed in the next commit because the design is to create the KOT at the client and the server by default without exchanging messages.